### ISSUE TITLE
Don't simply disable "submit" buttons

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-ui#1285]
Even when an account has no privilege to do some changes when browsing another account, or can't do certain operations due to no fee or etc, should be **possible to create proposals**, which is an important feature. 

If the buttons are simply disabled,

**Comment by sschiessl-bcp** [comment]
Creating a proposal also costs a fee, but certainly with priviliges. 

Do you have examples in mind where the user can't do the function because of that?

**Comment by abitmore** [comment]
For example, go to https://wallet.bitshares.org/#/account/committee-account, then click send, no matter what I fill in the table, the "send" button is always disabled. So how can I propose a transfer from committee-account to another account? Also, sell/buy buttons on the market page are all disabled. And buttons on other pages.

It seems that current UX assumes that one can only create a transaction when she owns the account showing on the up-right corner. Actually it is not true.

Proposal is a very useful and powerful feature in BitShares, although it need some time and efforts for new users to understand. We used to be able to propose proposals easily in UI (e.g. with version 160420). Browse to another account, then propose some actions from the account, are the best/simplest use cases for proposals.

Possible solution:
* add an option in settings, e.g. "don't disable buttons when browsing another account, to enable creation of proposals", unchecked by default. So new users won't get confused, and advanced users can make use of it.

**Comment by wmbutler** [comment]
I think we should allow buttons to be active when the source account is a valid source account for a proposal, certainly. Do we have an easy way of knowing if the user proposing the transaction holds adequate credentials to propose for the selected source account prior to the action?

If so, this should be fairly easy to accommodate.

**Comment by abitmore** [comment]
> I think we should allow buttons to be active when the source account is a valid source account for a proposal, certainly. Do we have an easy way of knowing if the user proposing the transaction holds adequate credentials to propose for the selected source account prior to the action?

If the wallet contains at least one account, it can be used to **propose** any transaction. Fees is another thing.

> We have another issue along these lines for buy operations on the exchange where I think it's important to all the user to set up and submit the buy and allow the user to authenticate as an intermediate step before seeing the confirmations screen.

It does need UX.

**Comment by wmbutler** [comment]
OK, so @abitmore as I understand it, any account is eligible to propose (even if it doesn't really have any authority... is this a true statement?

**Comment by abitmore** [comment]
True.

**Comment by wmbutler** [comment]
1 for UX and 1 for implementation

**Comment by ahdigital** [comment]
> OK, so @abitmore as I understand it, any account is eligible to propose (even if it doesn't really have any authority)... is this a true statement?

> True.

I think it would help to have a list of all possible proposals a user can perform on behalf of another users account. It'd be good to make sure we cover all bases.

As part of the #1213 task I've proposed an updated sitemap and suggested an overhaul of the accounts page precisely because it's difficult to quickly understand which account is being viewed and what actions can be performed.

**Comment by abitmore** [comment]
> ... a list of all possible proposals ...

All "operations" aka "actions".

**Comment by abitmore** [comment]
Proposals are not "one user performs on behalf of another user", they're "one user asks another user to perform some actions", the "another user" can accept or ignore or refuse the request.

**Comment by ahdigital** [comment]
Makes sense. I've clearly not used this functionality before, so my question is where does a user currently go to within the DEX to accept/ignore/refuse?

**Comment by abitmore** [comment]
@ahdigital see https://github.com/bitshares/bitshares-ui/issues/1284.

**Comment by ahdigital** [comment]
I'll try again later when I have more time but the proposed tab doesn't show for me when viewing my own account or the example account above: 'committee-account'.

**Comment by abitmore** [comment]
I think the tab is hidden when an account has no proposed transaction. You can propose one for yourself.

**Comment by abitmore** [comment]
I'd suggest 10 hours for implementation. No clue how many hours for UX.

**Comment by abitmore** [comment]
Since we need to change buttons **everywhere**, it's best if someone can list affected pages first (this takes time as well), then we can adjust the hours accordingly after seen how many things need to be done, and perhaps create new issues for each page or group of pages.

**Comment by MarkoPaasila** [comment]
I can propose a transfer from any account using the legacy send dialog but not the new one

**Comment by abitmore** [comment]
Possible UX for this issue:
* Add an option "enable proposal proposing" in settings, disabled by default, and a note "this is an advanced feature, if enabled, the UI won't perform certain checking for you" or something similar;
  * when the option is enabled by the user, disable all database state checking on every action page (e.g. check balance before starting a transfer, and etc).

**Comment by wmbutler** [comment]
"Enable proposal proposing". Surely we can do better than that. This will definitely require some planning hours.
